### ISSUE TITLE
Windows 10 (1803) - Entity Framework Error after updating Windows - “The incoming tabular data stream (TDS) remote procedure call > (RPC) protocol stream is incorrect”

### ISSUE THREAD

**Issue by divega** [Microsoft/dotnet#749]
_From @dotnetshadow on May 15, 2018 7:10_

Hi there, 

I found some of my queries are not working after updating to Windows 10 version 1803. I can confirm that the same code works on machines without this update. 

The error I keep getting is this:
```sql
System.Data.Entity.Core.EntityCommandExecutionException: An error occurred while executing the command definition. See the inner exception for details. ---> System.Data.SqlClient.SqlException: The incoming tabular data stream (TDS) remote procedure call (RPC) protocol stream is incorrect. Parameter 2 (""): Data type 0x00 is unknown.
```

Interestingly if I make the query non async then everything seems to work. Has something changed since Windows 10?

**Other users have also indicated having the same problem**
https://stackoverflow.com/questions/50340258/entity-framework-error-after-updating-windows-the-incoming-tabular-data-strea

### Further technical details
EF version: 6.2
Database Provider: Azure SQL Server
Operating system: Windows 10 (1803) - April 2018 update 
IDE: Visual Studio 2017 15.7

_Copied from original issue: aspnet/EntityFramework6#537_

**Comment by divega** [comment]
_From @chainbridgetech on May 15, 2018 12:53_

I can report that I received the same error after updating to 1803. A system restore back to 1709 has temporarily fixed the problem.

**Comment by divega** [comment]
_From @ErikEJ on May 15, 2018 12:56_

Sounds like a SqlClient bug, not an EF6 issue

**Comment by divega** [comment]
_From @dotnetshadow on May 15, 2018 13:6_

@ErikEJ Any idea where I can report this?

**Comment by divega** [comment]
_From @ErikEJ on May 15, 2018 13:35_

I am sure the EF team will know who to notify

**Comment by divega** [comment]
_From @chainbridgetech on May 15, 2018 19:21_

I have confirmed that this is specifically caused by .NET Framework 4.7.2. I manually installed .NET 4.7.2 on my Windows 10 1709 computer, and the error started up again.

**Comment by divega** [comment]
@dotnetshadow Thanks for reporting this. I am going to be moving this issue to https://github.com/Microsoft/dotnet/issues, which is where issues for SqlClient on .NET Framework are tracked. In the meantime, it would be great if you could provide a repro project.

**Comment by chainbridgetech** [comment]
I am still working on a simplified repro project, but I have found that if I change MultipleActiveResultSets from TRUE to FALSE in my connection string, the error stops.

**Comment by corivera** [comment]
@chainbridgetech Would you be able to make simple test tables in your repro project, or is this error actually data-dependent?

**Comment by chainbridgetech** [comment]
I created a local db in the project that exactly matches the Azure db (on connection string is commented out and can be flipped to test), but the error doesn't occur when connecting to the local db only when connecting to SQL Azure. It is not dependent on the data. The tables can be empty and the error occurs.

**Comment by corivera** [comment]
Would you be able to push up your repro project to Github then? If the tables can be empty, then I can just run the repro against my own Azure DB.

**Comment by chainbridgetech** [comment]
https://github.com/chainbridgetech/net472-rpc_error-repro -- I can send you the password if you want to try it against my test db (that doesn't have any private data).

**Comment by corivera** [comment]
No, that won't be necessary. Are there test table creation statements included in the project?

**Comment by chainbridgetech** [comment]
https://github.com/chainbridgetech/net472-rpc_error-repro/blob/master/Models/db_create.sql

**Comment by corivera** [comment]
Great, I'll look into it.

**Comment by chainbridgetech** [comment]
Were you able to replicate the exception using my repro project?

**Comment by corivera** [comment]
Yes, I was able to repro the issue. I'm investigating potential fixes.

**Comment by leonardobaggio** [comment]
I'm also facing this exact same issue at dev environment in two different machines (both running Windows version 1803). If you guys need any further technical details, please let me know.

**Comment by demodav** [comment]
I am facing the same issue.

**Comment by Joshuapwilley** [comment]
Also experiencing this issue. Another temporary fix appears to be removing `async` from the queries.

**Comment by Joshuapwilley** [comment]
@demodav You should make others aware that the method you referenced as a solution is only available via a third-party package. It is not part of Entity Framework.

**Comment by joylemicrosoft** [comment]
An update from Azure SQL Engineering team: Engineers have developed a SQL DB server side fix and are rolling out through the safe deployment procedure. ETA 24-48 hours.

**Comment by dotnetshadow** [comment]
I can confirm that the issue seems to be resolved, thank you for all your efforts, great work

**Comment by Joshuapwilley** [comment]
I can also confirm that the issue appears to be resolved as well.

**Comment by chainbridgetech** [comment]
For the server where I discovered the issue, the problem is fixed. However, for the server I created for the purposes of creating a repro project, I am still getting the error. I am going to assume for now that it simply hasn't received the fix yet, and I will recheck it tomorrow.

**Comment by chainbridgetech** [comment]
Azure support was able to confirm that the fix has not been rolled out to the server where I am still experiencing the error.

**Comment by bwaldron** [comment]
@joylemicrosoft: Is there a time where this fix has been rolled out to all servers?  We are still seeing the error with Windows 1803.

**Comment by petercarlin** [comment]
The fix has been rolled out to all SQL DB servers.  If you are seeing it at this point, please contact Microsoft support.

**Comment by leonardobaggio** [comment]
@petercarlin good to hear that these improvements for AE are going on ...

Sorry to write this here, probably this isn`t the right place. I'm currently in a migration  of some columns to Always Encrypted in a currently populated Azure SQL Database, and the main application that consumes it using Entity Framework. The overall process is being quite painful.

A lot of unexpected "Operand clash" errors in SELECT operations required us to recode many of our existing queries, especially the ones with .Include () method that EF uses UNION ALL Operator to deal with.

Anyway, I noticed in the middle of the last week that some error messages have changed, and it now specifies the column that is causing the error, etc ... My question is, is there a place like github or other community that can I follow and be aware of this improvements that are being made at the Always Encrypted functionality in SQL?

Thanks,
Leo

**Comment by petercarlin** [comment]
I do not know of a specific Always encrypted forum.  But if send me email at p e t e r c a @ Microsoft I will connect you with the program managers working on Always Encrypted who can work to keep you up to date.
